docs/vm/proposal.md

```diff
@@ -335,115 +311,363 @@ That isn't always possible.
+A debug table find style operation that just does something like counting the number of tables in a stream and returning that is fine.
+
+The important part is both demonstrating that we can execute the plan during the VM execution.
+This is in contrast to yield which gets implicitly executed after the main function has been executed.
+It is also important in this phase to demonstrate that we can detect when a plan node is split between multiple plans and should result in an error.
+
+More details about that situation are in the [table-find.md](../table-find.md) planning document.
+
+The following will be true at the end of this phase:
+
+* IR instructions exist that will execute a plan.
+* The IR instruction does not need to be useful or actually implement an existing `tableFind` function.
+* It should be an error to use a procedure spec in multiple disconnected plans.
+* Orphaned plan node children (ones that are included in a plan, but not actually materialized) should be detected as an error.
+
+### Value Type <a name="value-type"></a>
+
+This phase follows the creation of the initial VM.
+We will create a way to represent a value type in the IR and then have a corresponding generic value type that can represent any value in the VM.
+
+As part of this phase, additional `const` operations should be created that will create a value from a constant value.
+This should correspond with the various literal nodes in the semantic graph.
+
+Similarly, the IR generation will be capable of transforming literals to the `const` operation.
+
+The following will be true at the end of this phase:
+
+* A value type exists in the IR.
+* A value exists in the VM.
+* Types that are generic should resolve to the value type.
+* Constant operations are included in the IR.
+* Constant operations are implemented in the VM.
+
+### Value Operations
+
+This phase focuses on implementing the various builtin operations that are associated with manipulating values.
+This primarily includes binary operations but also includes unary operations, member operations, index operations, etc.
+
+The only thing this doesn't include are functions which will be handled separately.
 
-### Phase 1
+The following will be true at the end of this phase:
+
+* Operations manipulating values will be included in the IR.
+* Operations manipulating values will be implemented in the IR.
+
+### Table Find
+
+This phase continues the work from the previous spike by implementing the builtins associated with table find.
+These are presently `tableFind`, `getColumn`, and `getRecord`.
+
+The output of these should be the corresponding value from the [Value Type](#value-type) phase.
+
+### Functions
+
+This phase focuses on manifesting functions from function expressions.
+The functions can then be invoked later by a call expression.
+
+This phase does not include reifying generic types.
```

Comment:
If we go with, say WASM, not reifying may complicate this step since it effectively means we need dynamic implemented first or we can't type the functions.